packages/harpguru-core/src/components/covariant-menu/covariant-menu.tsx

```diff
@@ -1,92 +1,250 @@
+      />
+    )
+  }, [useGlobal])
 
-  const { harpKeyId } = activeHarpStrata
-  const nudgeHarpStrataByHarpKey = useNudgeHarpStrataByHarpKey()
-  const setHarpStrataByHarpKey = useSetHarpStrataByHarpKey()
-  const harpKeyOptionProps = {
-    title: 'Harp Key',
-    activeOptionId: harpKeyId,
-    orderedOptionIds: orderedPitchIds,
-    nudgeFunction: nudgeHarpStrataByHarpKey,
-    setFunction: setHarpStrataByHarpKey as (arg0: OptionIds) => void,
-  }
-  const harpKeyIsLocked = lockedCovariant === CovariantMembers.HarpKey
-  const lockHarpKey = () => {
-    setLockedCovariant(CovariantMembers.HarpKey)
-  }
+  const useHarpKeyLabel = useCallback(() => {
+    const [activeHarpStrata] = useGlobal('activeHarpStrata')
+    const { harpKeyId } = activeHarpStrata
+    return (
+      <OptionLabel
+        name={'Harp key'}
+        isLargeTitle={false}
+        value={getPitch(harpKeyId)}
+        alignItems={'center'}
+      />
+    )
+  }, [useGlobal])
 
-  const { pozitionId } = activeHarpStrata
-  const nudgeHarpStrataByPozition = useNudgeHarpStrataByPozition()
-  const setHarpStrataByPozition = useSetHarpStrataByPozition()
-  const pozitionOptionProps = {
-    title: 'Position',
-    activeOptionId: pozitionId,
-    orderedOptionIds: orderedPozitionIds,
-    nudgeFunction: nudgeHarpStrataByPozition,
-    setFunction: setHarpStrataByPozition as (arg0: OptionIds) => void,
-  }
-  const pozitionIsLocked = lockedCovariant === CovariantMembers.Pozition
-  const lockPozition = () => {
-    setLockedCovariant(CovariantMembers.Pozition)
-  }
+  const usePozitionLabel = useCallback(() => {
+    const [activeHarpStrata] = useGlobal('activeHarpStrata')
+    const { pozitionId } = activeHarpStrata
+    return (
+      <OptionLabel
+        name={'Position'}
+        isLargeTitle={false}
+        value={getPozition(pozitionId)}
+        alignItems={'center'}
+      />
+    )
+  }, [useGlobal])
 
-  const { rootPitchId } = activeHarpStrata
-  const nudgeHarpStrataByRootPitch = useNudgeHarpStrataByRootPitch()
-  const setHarpStrataByRootPitch = useSetHarpStrataByRootPitch()
-  const rootPitchOptionProps = {
-    title: 'Root Pitch',
-    activeOptionId: rootPitchId,
-    orderedOptionIds: orderedPitchIds,
-    nudgeFunction: nudgeHarpStrataByRootPitch,
-    setFunction: setHarpStrataByRootPitch as (arg0: OptionIds) => void,
-  }
-  const rootPitchIsLocked = lockedCovariant === CovariantMembers.RootPitch
-  const lockRootPitch = () => {
-    setLockedCovariant(CovariantMembers.RootPitch)
-  }
+  const useRootPitchLabel = useCallback(() => {
+    const [activeHarpStrata] = useGlobal('activeHarpStrata')
+    const { rootPitchId } = activeHarpStrata
+    return (
+      <OptionLabel
+        name={'Song key'}
+        isLargeTitle={false}
+        value={getPitch(rootPitchId)}
+        alignItems={'center'}
+      />
+    )
+  }, [useGlobal])
 
   const sizes = getSizes()
 
+  const optionPropsz = [
+    {
+      useTitle: useHarpKeyTitle,
```

Comment:
The body of all of these component hooks should be provided by a / some dedicated files.

How to break it up is the question. One thought that occurs to me is that it might be nice to make each of the `useCallback` callbacks be it's own isolated function which can be tested independently of the `useCallback` function.

I just need to be careful that the `useCallback` continues to work as expected when it's first parameter is a named function and the `useGlobal` is no longer visibly in scope. It should logically be fine.

Reply:
Done